easy-ui-react/src/DataGrid/ActionsCellContent.tsx

```diff
@@ -0,0 +1,64 @@
+import MoreVertIcon from "@easypost/easy-ui-icons/MoreVert";
+import React, { useCallback } from "react";
+import { Icon } from "../Icon";
+import { Menu } from "../Menu";
+import { Text } from "../Text";
+import { UnstyledPressButton } from "./UnstyledPressButton";
+import { useDataGridRow } from "./context";
+import {
+  ActionRowAction as ActionRowActionType,
+  MenuRowAction as MenuRowActionType,
+  RowAction,
+} from "./types";
+
+import styles from "./ActionsCellContent.module.scss";
+
+type ActionsCellContentProps = {
+  rowActions: RowAction[];
+};
+
+export function ActionsCellContent({ rowActions }: ActionsCellContentProps) {
+  return (
+    <span className={styles.ActionsCellContent}>
+      {rowActions.map((rowAction, i) =>
+        rowAction.type === "menu" ? (
+          <MenuRowAction key={i} rowAction={rowAction} />
+        ) : (
+          <ActionRowAction key={i} rowAction={rowAction} />
+        ),
+      )}
+    </span>
+  );
+}
+
+function MenuRowAction({ rowAction }: { rowAction: MenuRowActionType }) {
+  const { accessibilityLabel = "Actions" } = rowAction;
+  const { removeHover } = useDataGridRow();
+  const handleClick = useCallback(() => {
+    removeHover();
+  }, [removeHover]);
+  return (
+    <Menu>
+      <Menu.Trigger>
+        <UnstyledPressButton onPress={handleClick}>
+          <Text visuallyHidden>{accessibilityLabel}</Text>
+          <Icon symbol={MoreVertIcon} />
+        </UnstyledPressButton>
+      </Menu.Trigger>
+      {rowAction.renderMenuOverlay()}
```

Comment:
i understand the tension here. i originally had a design like this, in fact, but decided on deferring the entire overlay so that the consumer can have direct access to the props of the `Menu.Overlay` component (`disabledKeys`, `maxItemsUntilScroll`, `onAction`, `onClose`, `placement`, and `width`) without `DataGrid` having to take care of plumbing in the implementation